Monads/Option.cs

```diff
@@ -133,4 +133,138 @@ public class Some<T>(T value) : Option<T>(value) where T : class;
     /// </summary>
     /// <typeparam name="T">the type of the value that is missing</typeparam>
     public class None<T>() : Option<T>() where T : class;
+
+    /// <summary>
+    /// The monad type used to represent that there might be a value, but doesn't have to be as well.
+    /// </summary>
+    /// <remarks>Also provides simple shorthands for ways of manipulating with them.</remarks>
+    /// <typeparam name="T">the underlying value</typeparam>
+    public class ValueOption<T> : IEquatable<ValueOption<T>>
+        where T : struct
+    {
+        protected T? _value;
+
+        /// <summary>
+        /// Returns the underlying value
+        /// </summary>
+        /// <returns>the contained value</returns>
+        /// <exception cref="InvalidOperationException">there is no underlying value</exception>
+        public T Value => _value ?? throw new InvalidOperationException();
```

Comment:
Use null-member access operator `?.` to derefence `Value` instead
```suggestion
        public T Value => _value?.Value ?? throw new InvalidOperationException();
```

Reply:
nevermind, this doesn't exist